pkg/controllers/jobset_controller.go

```diff
@@ -237,16 +237,17 @@ func (r *JobSetReconciler) createHeadlessSvcIfNotExist(ctx context.Context, js *
 	// Check if service already exists. Service name is same as replicatedJob name.
 	// If the service does not exist, create it.
 	var headlessSvc corev1.Service
-	if err := r.Get(ctx, types.NamespacedName{Name: rjob.Name, Namespace: js.Namespace}, &headlessSvc); err != nil {
+	if err := r.Get(ctx, types.NamespacedName{Name: genSubdomain(js, rjob), Namespace: js.Namespace}, &headlessSvc); err != nil {
```

Comment:
`subdomain := genSubdomain(js, rjob)`

Reply:
Woops I already caught this and fixed it earlier but didn't push the changes I guess, my bad

Reply:
not yet addressed

Reply:
Sorry I misinterpreted the comment to mean the svc name should be the subdomain rather than the replicatedJob name, which was already the case. I've updated the code to store the subdomain in a variable and reuse it.